source/toolkits/S3CredentialStore.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file S3CredentialStore.h
+ *
+ * Singleton class to manage multiple S3 credentials for different workers.
```

Comment:
let's merge this comment with the class comment, so that we only have the comment on the `class` and no a general comment at the top of the file.